core/src/main/scala/pureconfig/DerivedWriters.scala

```diff
@@ -12,6 +12,7 @@ trait DerivedWriters extends DerivedWriters1 {
   implicit def deriveAnyVal[T, U](
     implicit
     ev: T <:< AnyVal,
+    generic: Generic[T],
```

Comment:
I did not know shapeless generated `Generic` instances for classes that are not case classes. Does this happen only because classes like `IntWrapper` are `AnyVal` instances?